services/arrow-flight/src/main/java/com/dremio/service/flight/impl/GetCatalogsResponseHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2017-2019 Dremio Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.dremio.service.flight.impl;
+
+import org.apache.arrow.flight.FlightProducer;
+import org.apache.arrow.flight.sql.FlightSqlProducer;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.util.Text;
+
+import com.dremio.common.utils.protos.QueryWritableBatch;
+import com.dremio.exec.proto.GeneralRPCProtos;
+import com.dremio.exec.proto.UserProtos;
+import com.dremio.exec.rpc.RpcOutcomeListener;
+import com.dremio.exec.work.protector.UserResponseHandler;
+import com.dremio.exec.work.protector.UserResult;
+
+/**
+ * {@link UserResponseHandler} implementation for {@link FlightWorkManager#getCatalogs}.
+ */
+class GetCatalogsResponseHandler implements UserResponseHandler {
```

Comment:
This class should extend CancellableUserResponseHandler, similar to how we have CreatePreparedStatementResponseHandler.

Reply:
You are right! 
I refactored CancellableUserResponseHandler to handle both cases (CreatePreparedStatement, GetCatalogs and will apply to all others) and deleted the subclasses CreatePreparedStatementResponseHandler and GetCatalogsResponseHandler.

Can you take a look now please? @jduo